src/js/firstparties/google-search.js

```diff
@@ -1,8 +1,17 @@
 /* globals findInAllFrames:false */
+
+// bool for whether or not in firefox for android
+let firefox_android = (navigator.userAgent.indexOf('Firefox') > -1 && navigator.userAgent.indexOf('Android') > -1);
```

Comment:
We should try to avoid user agent matching whenever we can as it is inherently brittle. The good news is that most of the time there is no need for it. Generally [testing for features](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) is a much better idea.

Reply:
I think for most cases not relying on user agent sniffing is correct, but this might be a rare example of when it isn’t. 

We aren’t relying on a specific feature, but rather how Google search results links are structured when viewed through Firefox for Android browser. For example, we can’t reliably lean on device width or touchscreen features here, since there would be some inaccuracies with large tablets and/or touchscreen laptops. The biggest commonality across the different devices that I can think of is there being the presence of both ‘Firefox’ and ‘Android’ somewhere in the user agent string.

Reply:
The "feature" here we should test is what the search result links look like. Test on desired features/functionality.